src/pydvl/influence/general.py

```diff
@@ -105,11 +96,11 @@ def _calculate_influences_up(
 
 def _calculate_influences_pert(
     model: TwiceDifferentiable,
-    x: "NDArray",
-    y: "NDArray",
-    influence_factors: "NDArray",
+    x: TensorType,
```

Comment:
See my comment below about union types and how they fail to catch mixed argument types. If you want to accept both values, you can use `@overload` for the benefit of the type checker. Or a generic function with a TypeVar constrained to the two types (but I'm not sure if that would catch the mixed type problem).

Here's an example for the first:

```python
@overload
def _calculate_influences_pert(model: NumpyTwiceDifferentiable, x: NDArray[np.float_], blahblah) -> NDArray[np.float_]:
    ...
@overload
def _calculate_influences_pert(model: TorchTwiceDifferentiable, x: torch.Tensor, blahblah) -> torch.Tensor:
    ...

def _calculate_influences_pert(model: Union[], x: Union[]...) -> Union[...]:
    # schtuff

Reply:
should be solved with the new changes.